### ISSUE TITLE
Cannot add products to cart with REST call

### ISSUE THREAD

**Issue by keharper** [magento/inventory#1513]
This might be related to [Issue-1504](https://github.com/magento-engcom/msi/issues/1504)

### Preconditions
1. 2.3-develop branch of MSI
2. You need to supply an admin auth token for steps 1-8. To generate one, use `POST http://<host>/rest/uk/V1/integration/admin/token`
. Payload:
```
{
"username": "admin",
"password": "<password>"
}
```
### Steps to reproduce

1. Create two websites / stores / store views.

Website UK Site code: uk_site
UK Website Store code: uk_website_store
UK Store View code: uk

Website US Site code: us_site
US Website Store code: us_website_store
US Store View code: us

2. Create several sources with `POST http://<host>/rest/all/V1/inventory/sources`

Payloads:

```
{
   "source" : {
      "name" : "London Warehouse",
      "source_code" : "uk_warehouse",
      "postcode" : "WC2N 5DU",
      "enabled" : true,
      "contact_name" : "Ethan Carter",
      "country_id" : "GB",
      "city" : "London"
   }
}
```

```
{
   "source" : {
      "name" : "London Flagship Store",
      "source_code" : "uk_store",
      "postcode" : "SW1A 1AB",
      "enabled" : true,
      "contact_name" : "Ralf Schneider",
      "country_id" : "GB",
      "city" : "London"
   }
}
```
 ```
{
   "source" : {
      "name" : "Berlin Warehouse",
      "source_code" : "de_warehouse",
      "postcode" : "10115",
      "enabled" : true,
      "contact_name" : "Ralf Schneider",
      "country_id" : "DE",
      "city" : "Berlin"
   }
}
```
```
{
   "source" : {
      "name" : "New York Warehouse",
      "source_code" : "us_warehouse",
      "postcode" : "11368",
      "enabled" : true,
      "contact_name" : "Fred Wilson",
      "country_id" : "US",
      "city" : "New York"
   }
}
```
3. Add a Stocks with `POST http://<host>/rest/all/V1/inventory/stocks` 
Payload:
```
{
  "stock": {
    "name": "UK Stock",
    "extension_attributes": {
      "sales_channels": [
        {
          "type": "website",
          "code": "uk"
        }
      ]
    }
  }
}
```

4. Use `POST http://<host>/rest/all/V1/inventory/stock-source-links` to link the stocks and sources. Payload: 

```
{
   "links" : [
      {
         "source_code" : "uk_warehouse",
         "stock_id" : "2",
         "priority" : 1
      },
      {
        "source_code" : "uk_store",
        "stock_id" : "2",
        "priority" : 2
      },
      {
        "source_code" : "de_warehouse",
        "stock_id" : "2",
        "priority" : 3
      },
       {
        "source_code" : "us_warehouse",
        "stock_id" : "3",
        "priority" : 1
      },
      {
         "source_code" : "uk_warehouse",
         "stock_id" : "3",
         "priority" : 2
      }
   ]
}
```
5. Create a simple product with `POST http://<host>/rest/all/V1/products` . Payload:
```
{
	"product": {
    "name": "Simple Product 1",
		"sku": "sp1",
		"attribute_set_id": 4,
		"price": 5,
		"status": 1,
		"visibility": 4,
		"type_id": "simple",
		"weight": 0.8,
		"extension_attributes": {
			"website_ids": ["1", "2", "3"]
		}
	}
}
```
6. Unassign the product from the default source with `POST http://<host>/rest/all/V1/inventory/source-items-delete`. Payload:
```
{
	"sourceItems": [{
		"sku": "sp1",
		"source_code": "default"
  }]
}
```
7. Assign the product to custom sources with `POST http://<host>/rest/all/V1/inventory/source-items`. Payload:
```
{
  "sourceItems": [{
	"sku": "sp1",
	"source_code": "uk_warehouse",
  "quantity": 10,
  "status": 1
	},
  {
  "sku": "sp1",
	"source_code": "uk_store",
  "quantity": 5,
  "status": 1
  },
  {
  "sku": "sp1",
	"source_code": "de_warehouse",
  "quantity": 25,
  "status": 1
  },
  {
  "sku": "sp1",
	"source_code": "us_warehouse",
  "quantity": 60,
  "status": 1
  }]
}
```

8. Create a `uk` store customer with `POST http://<host>/rest/uk/V1/customers`. Payload:
```
{
   "customer" : {
      "lastname" : "Doe",
      "firstname" : "Jane",
      "email" : "jdoe@example.com",
      "addresses" : [
         {
            "defaultBilling" : true,
            "defaultShipping" : true,
            "firstname" : "Jane",
            "lastname" : "Doe",
            "region" : {
               "regionCode" : "NY",
               "regionId" : 43,
               "region" : "New York"
            },
            "countryId" : "US",
            "postcode" : "10755",
            "city" : "Purchase",
            "street" : [
               "123 Oak Ave"
            ],
            "telephone" : "512-555-1111"
         }
      ]
   },
   "password" : "Password1"
}
```
9. Generate a token for the customer with `POST http://<host>/rest/uk/V1/integration/customer/token`
. Do not include an admin auth token in the headers. Payload:
```
{
"username": "jdoe@example.com",
"password": "Password1"
}
```

10. Create a cart on behalf of the customer. Use the customer token in the header of the call:
`POST http://<host>/rest/uk/V1/carts/mine` (no payload)

11. Add the product you created earlier to the cart using `POST http://<host>/rest/uk/V1/carts/mine/items`. Payload:
```
{
  "cartItem": {
    "sku": "sp1",
    "qty": 20,
    "quote_id": "1"
  }
}
```

### Expected result
The items are added to the cart. 

### Actual result
"message": "Product that you are trying to add is not available."

**Comment by maghamed** [comment]
After step 7 please make sure that the product `sp1 ` is available on `uk`storefront

**Comment by VitaliyBoyko** [comment]
After setting the correct website code in step 3 I've got the expected result. Preventing save stock with incorrect website code resolved during an investigation of #1504, added in PR #1514.
Closing this one.
